File: R/count.R
#' Dispatch a count to some Cassandra table
#'
#' @param session The Cassandra session we will dispatch the query in
#' @param table The table to which to connect
#'
#' @export
cql_count <- function(sc, session, table) {
  sparklyr::invoke_static(sc, "CQLConnect.Count", "get_count", session, table)
}

File: java/Count.scala
package CQLConnect

import com.weather.scalacass.ScalaSession


object Count {

  def get_count(session: ScalaSession, table: String): Long = {

    val keyspace = session.keyspace

    val count_query =
      s"""SELECT
         |  count(*)
         |FROM $keyspace.$table
         |""".stripMargin

    session.rawSelect(count_query).next().getLong("count")

  }

}
